packages/react-components/src/atoms/Typeahead.tsx

```diff
@@ -81,6 +81,12 @@ const Typeahead: FC<TypeaheadProps> = ({
       customValidationMessage,
       getValidationMessage,
     );
+  const onNewValue = (newValue: string) => {
+    setInputValue(newValue);
+    onChange(newValue);
+    // Hack to re-validate once the selected value has been put in the state of the input field
+    setTimeout(validationTargetProps.onBlur, 0);
```

Comment:
Yeah, this will be on every keypress which isn't ideal. I am hoping we can move to something more like twitter which remove the error message while typing but puts it back after a second. I think the validation logic is complicated and something that we need to carefully refine and outside the scope of this ticket.